silkworm/state/intra_block_state.hpp

```diff
@@ -51,9 +51,9 @@ class IntraBlockState {
     IntraBlockState(const IntraBlockState&) = delete;
     IntraBlockState& operator=(const IntraBlockState&) = delete;
 
-    explicit IntraBlockState(db::Buffer& db) noexcept : db_{db} {}
+    explicit IntraBlockState(db::StateBuffer& db) noexcept : db_{db} {}
 
-    db::Buffer& db() { return db_; }
+    db::StateBuffer& db() const { return db_; }
```

Comment:
I wouldn't mark this method as const since it returns a non-const reference.

Reply:
Oh yes, my bad